flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -686,7 +701,14 @@ public void createTemporaryTable(
                         }
                         return v;
                     } else {
-                        final CatalogBaseTable resolvedTable = resolveCatalogBaseTable(table);
+                        ResolvedCatalogBaseTable<?> resolvedTable =
+                                managedTableListener.notifyTableCreation(
+                                        getCatalog(objectIdentifier.getCatalogName()).orElse(null),
+                                        objectIdentifier,
+                                        resolveCatalogBaseTable(table),
+                                        true,
```

Comment:
aren't the properties "temporary" and "catalog = null" related? I don't think we need the temporary flag. we can still add it on the internal API if necessary.

Reply:
Our temporary tables can be catalog-owned, I remember this is why we do: https://issues.apache.org/jira/browse/FLINK-21185

Reply:
FYI: We also do `org.apache.flink.table.catalog.CatalogManager.TableLookupResult#isTemporary`. So the temporary table might be catalog owned during creation but will look like a non-temporary one currently in the factory.

But I'm fine with the change.

Reply:
I got it. Here is for temporary managed table.